requests/request_body.go

```diff
@@ -29,9 +30,30 @@ type RequestBodyValidator interface {
 	ValidateRequestBodyWithPathItem(request *http.Request, pathItem *v3.PathItem, pathValue string) (bool, []*errors.ValidationError)
 }
 
+type configOptions struct {
+	regexEngine jsonschema.RegexpEngine
+}
+
+// Option supports the 'Options Pattern' to define the behavior of a RequestBodyValidator
+type Option func(options *configOptions)
+
+// WithRegexEngine allows for a custom regular expression engine to be used during validation.
+func WithRegexEngine(engine jsonschema.RegexpEngine) Option {
+	return func(rbv *configOptions) {
+		rbv.regexEngine = engine
+	}
+}
+
 // NewRequestBodyValidator will create a new RequestBodyValidator from an OpenAPI 3+ document
-func NewRequestBodyValidator(document *v3.Document) RequestBodyValidator {
-	return &requestBodyValidator{document: document, schemaCache: &sync.Map{}}
+func NewRequestBodyValidator(document *v3.Document, opt ...Option) RequestBodyValidator {
+
+	cfg := configOptions{} // Default Options
+	for _, o := range opt {
+		o(&cfg)
+	}
+
+	return &requestBodyValidator{configOptions: cfg, document: document, schemaCache: &sync.Map{}}
```

Comment:
Do you have a need for a configOptions here?

I mean the regexpEngine could be inside the struct of requestBodyValidator


```suggestion
	rbv := &requestBodyValidator{configOptions: cfg, document: document, schemaCache: &sync.Map{}}

	for _, opt := range opts {
		o(rbv)
	}

	return rbv
```

```go
 type requestBodyValidator struct { 
          regexEngine jsonschema.RegexpEngine
          document    *v3.Document 
          schemaCache *sync.Map 
  }

// Option supports the 'Options Pattern' to define the behavior of a RequestBodyValidator 
 type Option func(rbv *requestBodyValidator) 
  
 // WithRegexEngine allows for a custom regular expression engine to be used during validation. 
 func WithRegexEngine(engine jsonschema.RegexpEngine) Option { 
         return func(rbv *requestBodyValidator) { 
                 rbv.regexEngine = engine 
         } 
 }
```

Reply:
I went in this direction because the `requests` package has public functions that are not associated with the internal `requestBodyValidator` type - eg ValidateRequestSchema.

Technically speaking I guess I could hijack that struct and use it in a context it presumably wasn't designed for but that seemed a bit dubious.

I did go back-and-forth on that, i can hijack if that's deemed appropriate. :-)

Reply:
I would say it's more a question for a maintainer: @emilien-puget maybe